docs/docs/detectors/14-iterators-over-indexing.md

```diff
@@ -1,19 +1,24 @@
-# Iterators-over-indexing
+# Iterators over indexing
 
-### What it does
+## Description 
 
-It warns if the for loop uses indexing instead of iterator. If the indexing goes to length it will not raise a warning.
+- Category: `Best practices`
+- Severity: `Enhancement`
+- Detector: [`iterators-over-indexing`](https://github.com/CoinFabrik/scout-soroban/tree/main/detectors/iterators-over-indexing)
+- Test Cases: [`iterators-over-indexing-1`](https://github.com/CoinFabrik/scout-soroban/tree/main/test-cases/iterators-over-indexing/iterators-over-indexing-1)
 
-### Why is this bad?
+In Rust, sequences can be traversed using iterators or direct indexing. However, the least efficient way is through direct indexing.
 
+## Why is this bad? 
 
-Accessing a vector by index is slower than using an iterator. Also, if the index is out of bounds, it will panic.
+When you iterate over a data structure with fixed limits, you might exceed the limit, causing a panic in the contract and other potential errors.
```

Comment:
Una sugerencia para que quede un poquito más claro: "When you iterate over a data structure with fixed limits in a Soroban smart contract, exceeding those limits can cause the contract to panic, potentially leading to errors or unexpected behavior."